src/Language/PureScript/Ide/CaseSplit.hs

```diff
@@ -119,11 +120,17 @@ addClause s wca =
         " = ?" <> (T.strip . T.pack . runIdent $ fName)
   in [s, template]
 
-parseType' :: String -> Type
-parseType' s = let (Right t) = do
-                     ts <- lex "" s
-                     runTokenParser "" (parseType <* P.eof) ts
-               in t
+parseType' :: (MonadError PscIdeError m) =>
+              String -> m Type
+parseType' s = let
+                 parseResult = do
+                   ts <- lex "" s
+                   runTokenParser "" (parseType <* P.eof) ts
+               in
+                 case parseResult of
```

Comment:
This is a type supplied by the user so there is no file involved. The oneliner seems simple enough.